File: leetcode/344.reverse-string/ts/reversestring.ts
// N, 1
const reverseString = (s: string[]): void => {
    let start = 0
    let end = s.length - 1

    while (start < end) {
        const temp = s[start]
        s[start] = s[end]
        s[end] = temp

        start++
        end--
    }
};

export default reverseString

File: leetcode/344.reverse-string/cpp/reversestring.cpp
#include <vector>

class Solution
{
public:
    void reverseString(std::vector<char> &characters)
    {
        if (characters.empty())
            return;

        auto itStart = characters.begin();
        auto itEnd = characters.end() - 1;

        while (itStart < itEnd)
        {
            auto temp = *itStart;
            *itStart = *itEnd;
            *itEnd = temp;

            ++itStart;
            --itEnd;
        }
    }
};
